src/swarm-tui/Swarm/TUI/Controller.hs

```diff
@@ -1264,8 +1265,8 @@ tabComplete CompletionContext {..} names em theRepl = case theRepl ^. replPrompt
 
   possibleWords =
     names <> case ctxCreativeMode of
-      True -> reservedWords
-      False -> filter (`notElem` creativeWords) reservedWords
+      True -> S.toList reservedWords
+      False -> filter (`notElem` creativeWords) (S.toList reservedWords)
```

Comment:
This looks quadratic?  Convert `creativeWords` to a set and use `difference`

Reply:
Oops, good point.  I'll include this change with #1851.